internal/sync/mutex_map.go

```diff
@@ -0,0 +1,65 @@
+package sync
+
+import "sync"
+
+// RWMutexMap is a concurrency-safe map, using a sync.RWMutex to lock a backing
+// map when accessing values.
+type RWMutexMap[T any] struct {
```

Comment:
I guess that you already checked Go version on RHEL 8 build? Just in case

Reply:
I believe as part of this commit, I changed the Go version in go.mod to 1.18. There is no plan to backport this version of yggdrasil to RHEL 8, so RHEL 8 having Go < 1.18 is not an issue.

Reply:
Then it's ok :-)